### DISCUSSION TITLE
python leaves behind pip installed files?

### DISCUSSION THREAD

**Discussion by rdp** [Homebrew/discussions#1314] [Everyday usage]
As a note if I do
$ brew install python3
$ pip3 install something
$ brew remove python3

it still leaves "somethings" behind.  Since it installs pip3, maybe it could clean up more?  Just a thought.

**Comment by carlocab** [comment]
`brew` doesn't know what you install with `pip3` (or whether something it finds was installed by a `pip3` bundled with Homebrew Python or something else entirely), so it's pretty difficult for `brew` to clean up more than it already does.

**Comment by rdp** [comment]
Since python by default "includes" pip, maybe it could run some kind of pipe "remove all" on its way out... https://stackoverflow.com/questions/11248073/what-is-the-easiest-way-to-remove-all-packages-installed-by-pip just an idea.  Though people who use pip are usually more hardcore so kind of know what they're doing...

**Reply by rdp** [reply]
Or maybe at least a warning message... :)

**Reply by carlocab** [reply]
If I install Python some other way (say, the Python.org installer), does it do either of those things? That is, uninstall everything you `pip3 install`ed, or show a warning message.

If it does, we can perhaps look into doing one of those things. If it doesn't, I'm not sure we should do either of those.
